hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AtttributeResolver.kt

```diff
@@ -0,0 +1,61 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch
+
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.TypeMatcher
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.genericType
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.memberProperties
+
+typealias Attributes = Map<String, KClass<*>>
+
+@Configuration
+class AttributeResolver {
+  @Bean
+  fun attributes(): Attributes = getAttributes(Prisoner::class)
+}
+
+internal fun getAttributes(kClass: KClass<*>): Attributes {
+  val map = mutableMapOf<String, KClass<*>>()
+  kClass.memberProperties.forEach { prop ->
```

Comment:
never a great fan of `forEach`, could `findTypes` return entries instead so you could use `map`?

Reply:
answering my own question - `fold` would work better
```kotlin
internal fun getAttributes(kClass: KClass<*>): Attributes =
  kClass.memberProperties.fold( mutableMapOf<String, KClass<*>>()) { map, prop ->
    findTypes(prop, map)
    map
  }.toMap()
```
if you got `findTypes` to return `map` then the extra `map` line could be removed too